python/kserve/pyproject.toml

```diff
@@ -53,35 +53,20 @@ pandas = ">=1.3.5"
 pydantic = ">1.0,<3"
 pyyaml = "^6.0.0"
 
-# Storage dependencies. They can be opted into by apps.
-requests = { version = "^2.20.0", optional = true }
-google-cloud-storage = { version = "^2.3.0", optional = true }
-azure-storage-blob = { version = "^12.10.0", optional = true }
-azure-storage-file-share = { version = "^12.7.0", optional = true }
-azure-identity = { version = "^1.8.0", optional = true }
-boto3 = { version = "^1.21.0", optional = true }
-
 # Logging dependencies. They can be opted into by apps.
-asgi-logger = { version = "^0.1.0", optional = true, python = ">3.8.0,<3.11" }
+asgi-logger = { version = "^0.1.0", optional = true, python = ">3.8.0,<3.12" }
 
 # Ray dependency
 ray = { version = "~2.10.0", extras = ["serve"] }
 
 [tool.poetry.extras]
-storage = [
```

Comment:
We can still include the storage package as an optional dependency , So that it won't be a breaking change. This will give users some time to migrate to the new storage package. Maybe then we can remove it from Kserve sdk.

Reply:
do you mean keep storage module around for a while?

Reply:
added back the storage package inside `python/kserve`.

Reply:
@sivanantha321  If I add this to the `kserve's` _pyproject,toml_:

```yaml
# Storage dependencies. They can be opted into by apps.
storage = { path = "../storage", extras = ["storage"], optional = true, develop = true }
```

The dependencies will be updated to:

```
--- a/python/pmmlserver/poetry.lock
+++ b/python/pmmlserver/poetry.lock
@@ -1148,7 +1148,7 @@ uvicorn = {version = "^0.21.1", extras = ["standard"]}
 [package.extras]
 logging = ["asgi-logger (>=0.1.0,<0.2.0)"]
 ray = ["ray[serve] (>=2.10.0,<2.11.0)"]
-storage = ["azure-identity (>=1.8.0,<2.0.0)", "azure-storage-blob (>=12.10.0,<13.0.0)", "azure-storage-file-share (>=12.7.0,<13.0.0)", "boto3 (>=1.21.0,<2.0.0)", "google-cloud-storage (>=2.3.0,<3.0.0)", "huggingface-hub (>=0.24.5,<0.25.0)", "requests (>=2.20.0,<3.0.0)"]
+storage = ["storage[storage] @ file:///Users/fspolti/data/dev/sources/kserve/python/storage"]
```

Reply:
it should be something like this
```
storage = { path = "../storage", optional = true, develop = true }

[tool.poetry.extras]
storage = [
"storage"
]

```

Reply:
That was what I did.
https://github.com/spolti/kserve/commit/c54fcd37464a1a32f08f596b9081c93c175f523b

Reply:
@sivanantha321 this line specifically:
https://github.com/spolti/kserve/commit/c54fcd37464a1a32f08f596b9081c93c175f523b#diff-ad2d8c87e4697eeeb1827029aaf92553c8ce992e2a65850374e3d1448e27f144R658